File: solutions/ruby/matrix/1/matrix.rb
=begin
Write your code for the 'Matrix' exercise in this file. Make the tests in
`matrix_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/matrix` directory.
=end

class Matrix
  def initialize(matrix)
    @matrixRows = matrix.split("\n")
  end

  def rows
    @rows = []
    for row in @matrixRows
      @rows.push(row.split.map { |el| el.to_i })
    end
    @rows
  end

  def columns
    @columns = []
    rows[0].length.times { @columns.push([]) }
    for row in rows
      row.length.times { |i| @columns[i].push(row[i]) }
    end
    @columns
  end
end

File: solutions/python/matrix/1/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        rowsString = matrix_string.split("\n")
        self.rows = [[] for x in rowsString]
        for i in range(len(rowsString)):
            self.rows[i] = [int(x) for x in rowsString[i].split(" ")]
        self.columns =[[] for x in range(len(self.rows[0]))]
        for r in self.rows:
            for i in range(len(r)):
                self.columns[i].append(r[i])
    def row(self, index):
        return self.rows[index-1]

    def column(self, index):
        return self.columns[index-1]